google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/spi/v2/HttpBigQueryRpc.java

```diff
@@ -130,6 +130,21 @@ private void validateRPC() throws BigQueryException, IOException {
   public Dataset getDataset(String projectId, String datasetId, Map<Option, ?> options) {
     try {
       validateRPC();
+      Integer accessPolicyVersion = null;
+      for (Map.Entry<Option, ?> entry : options.entrySet()) {
+        if (entry.getKey() == Option.ACCESS_POLICY_VERSION && entry.getValue() != null) {
+          accessPolicyVersion = (Integer) entry.getValue();
+        }
+      }
+      if (accessPolicyVersion != null) {
+        return bigquery
```

Comment:
nit:  I worry here is if there's a divergence between this chain and the default execute().  Is it worth doing a single chain of builders with a conditional?  Same with this pattern in the other RPCs (create, patch, etc).

Reply:
Yes that approach seems cleaner. I updated the conditional to toggle only the access policy version instead of creating an entirely new RPC.